docs/proposals/artifact-gc-proposal.md

```diff
@@ -0,0 +1,69 @@
+# Proposal for Artifact Garbage Collection
+
+## Introduction
+
+The motivation for this is to enable users to automatically have certain Artifacts specified to be automatically garbage collected.
+
+Artifacts can be specified for Garbage Collection at different stages: `OnWorkflowCompletion`, `OnWorkflowDeletion`, `OnWorkflowSuccess`, `OnWorkflowFailure`, or `Never`
+
+## Proposal Specifics
+
+### Workflow Spec changes
+
+1. `WorkflowSpec` has an `ArtifactGCStrategy`, which is the default for all artifacts: can be `OnWorkflowCompletion`, `OnWorkflowDeletion`, `OnWorkflowSuccess`, `OnWorkflowFailure`, or `Never`
+2. Artifact has an `ArtifactGCStrategy` (only really applies to Output Artifacts), which can override the setting for `WorkflowSpec`
+3. `WorkflowSpec` can add specification of Service Account and/or Annotations to be used for the Garbage Collection pods, to enable their access to the storage.
+
+### Workflow Status changes
+
+1. Artifact has a boolean `Deleted` flag
+2. `WorkflowStatus.Conditions` can be set to `ArtifactGCError`
+
+### Proposal Options
+
+These [slides](../assets/artifact-gc-proposal.pptx) go over the trade offs in options that were presented to the Argo Contributor meeting on 7/12/22.
+
+### Decision
+
+Following the meeting, these were the decisions:
+
+We will go with Option 2 from that presentation. We'll have one Pod that runs in the user's namespace and deletes all artifacts pertaining to an individual Garbage Collection strategy. Since `OnWorkflowSuccess` happens at the same time as `OnWorkflowCompletion` and `OnWorkflowFailure` also happens at the same time as `OnWorkflowCompletion`, we can consider consolidating these GC Strategies together.
```

Comment:
Can we document the two options, e.g. in an "alternatives considered" section so that readers don't have to go through the slides?

Reply:
sure